src/ConciseCasts.h

```diff
@@ -55,42 +55,71 @@ inline Expr u8(Expr e) {
     return cast(UInt(8, e.type().lanes()), e);
 }
 
+namespace Internal {
+
+inline Expr signed_saturating_cast(Expr e, int bits) {
+    if (e.type().element_of() == Int(bits)) {
+        return e;
+    }
+    if (bits <= e.type().bits()) {
```

Comment:
I think there's a bit of a corner case that happens to work now on accident: e being a float. I think this would work today because we only have saturating casts up to 32 bits, and the smallest float we have today is 32 bit. But, we might have float 16 at some point in the future.

Either way, I think it would be a bit more clear to add a condition to this if, || e.type().is_float().